Jasmine/Grid/ViewModels/GridViewModel.swift

```diff
@@ -1,3 +1,113 @@
-class GridViewModel: GameViewModel {
+import Foundation
 
+class GridViewModel: GridGameEngineProtocol {
+    /// Stores the grid data that will be used to display in the view controller.
+    private(set) var gridData: [Coordinate : String] = [:]
+    /// The delegate that the View Controller will conform to in some way, so that the Game Engine
+    /// View Model can call.
+    weak var delegate: GridGameViewControllerDelegate?
+    /// Specifies the remaining time left in the game.
+    private(set) var remainingTimeLeft: TimeInterval = 0.0
+    /// Specifies the total time allowed in the game.
+    private(set) var totalTimeAllowed: TimeInterval = Constants.Grid.time
+    /// Specifies the current score of the game. If the game has not started, it will be the initial
+    /// displayed score.
+    private(set) var currentScore: Int = 0
+
+    /// Tells the view model that the game has started.
+    func startGame() {
+        self.delegate?.notifyGameStatus(with: .inProgress)
+
+        remainingTimeLeft = Constants.Grid.time
+
+        Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { timer in
```

Comment:
startGame has nothing to do with scheduling timer. so abstract this method out, along with all the time operations